File: jsoup/src/single_page/alberta/Alberta.java
package single_page.alberta;

import base.Field;
import base.SinglePageScraper;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.util.Arrays;
import java.util.List;

public class Alberta extends SinglePageScraper {

    public Alberta(String country, String university, String department, String url) {
        super(country, university, department, url);
    }

    public static void main(String[] args) {
        Long t1 = System.currentTimeMillis();

        String country = "Ca";
        String university = "alberta";
        String department = "CS";
        String url = "https://www.ualberta.ca/computing-science/faculty-and-staff/faculty";
        new Alberta(country, university, department, url);

        System.out.println("Time: " + (System.currentTimeMillis() - t1));
    }

    @Override
    protected void createJson() {
        Elements tableRows = document.select("table tbody tr");
        for (Element row : tableRows) {
            try {
                String name = row.select("a:first-child").text();

                String email = row.select("td:last-child a").attr("href");

                String link = row.select("a:first-child").attr("href");

                List<String> researchInterests = arrayToStringWithTrim(row.select("td:nth-last-child(2)").text().split(","));

                Field field = new Field(name, email, link, researchInterests);
                fields.add(field);
            } catch (Exception e) {
                exception.put(document.title(), e.getMessage());
            }
        }
    }

    @Override
    protected void setDir() {
        dir = "src/single_page/alberta/";
    }
}

File: beautifulsoup/single-page/alberta/alberta.py
from base.Field import Field
from base.SinglePageScraper import SinglePageScraper
from base.PageScraper import current_time_millis


class Alberta(SinglePageScraper):
    def __init__(self, url, university, department):
        super().__init__(url, university, department)

    def create_json(self):
        for row in self.soup.select('table tbody tr'):
            try:
                f = Field(
                          row.select('td a')[0].text.strip(),
                          row.select('td a')[0]['href'].strip(),
                          row.select('td')[2].text.strip().split(','),
                          row.select('td a')[1]['href'].strip())
                self.res.append(f)
            except Exception as e:
                self.exception += str(e).strip()


if __name__ == '__main__':
    t1 = current_time_millis()

    url = 'https://www.ualberta.ca/computing-science/faculty-and-staff/faculty'
    university = 'alberta'
    department = 'CS'
    Alberta(url, university, department)

    print('Time: {}'.format(current_time_millis() - t1))
